pkg/controller/sharding/sharding_utils.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package sharding
+
+import (
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	"github.com/apecloud/kubeblocks/pkg/constant"
+)
+
+func GenShardNameList(shardingSpec *appsv1alpha1.ShardingSpec) []string {
+	compNameList := make([]string, 0)
+	if shardingSpec == nil {
+		return compNameList
+	}
+	for i := 0; i < int(shardingSpec.Shards); i++ {
+		compNameList = append(compNameList, constant.GenerateShardName(shardingSpec.Name, i))
```

Comment:
in comparison to ordinal, it is preferable to use a randomly generated string

Reply:
> in contrast to ordinal, it is recommended to use a randomly generated string with 2 or 4 characters

done, generated a random string with 3 characters